firebase-perf/src/main/java/com/google/firebase/perf/config/DeviceCacheManager.java

```diff
@@ -228,10 +229,11 @@ public Optional<Float> getFloat(String key) {
 
     try {
       // Default value should never be used because key existence is checked above.
-      return Optional.of(sharedPref.getFloat(key, 0.0f));
+      long defaultValue = Double.doubleToLongBits(0.0);
```

Comment:
Can you leave some comments for the reason why we are typecasting from Double to Long and viceversa below? It looks a bit weird at first, but I think this is done for a good reason!

Reply:
Done.